code/go/internal/validator/spec.go

```diff
@@ -54,5 +58,11 @@ func (s Spec) ValidatePackage(pkg Package) ValidationErrors {
 		return errs
 	}
 
-	return rootSpec.validate(pkg.Name, pkg.RootPath)
+	// Syntactic validations
+	errs = rootSpec.validate(pkg.Name, pkg.RootPath)
+
+	// Semantic validations
+	errs = append(errs, semantic.ValidateKibanaObjectIDs(pkg.RootPath)...)
```

Comment:
What do you think about transforming this line a bit. I'm thinking about the following form:

```go
var rules []RuleFunction
rules := append(rules, semantic.ValidateKibanaObjectIDs,
    semantic.ValidateKibanaFoo,
    semantic.ValidateKibanaFoobar,
    semantic.ValidateKibanaFoobaz,
)

err := rules.Validate(pkg.RootPath)
```

Reply:
I implemented something like this in e0aa080. Let me know what you think.